tests/tests/test-xcm/test-mock-hrmp.ts

```diff
@@ -2260,3 +2268,79 @@ describeDevMoonbeam("Mock XCM - receive horizontal transact ETHEREUM", (context)
     expect(await contractDeployed.methods.count().call()).to.eq("1");
   });
 });
+
+describeDevMoonbeam("Mock XCM - receive horizontal suspend", (context) => {
+  before("Should receive a suspend channel", async function () {
+    // We first simulate a reception for suspending a channel from parachain 1
+    const xcmpFormat: XcmpMessageFormat = context.polkadotApi.createType(
+      "XcmpMessageFormat",
+      "Signals"
+    ) as any;
+    const receivedMessage = context.polkadotApi.createType("u8", 0) as any;
+    const totalMessage = [...xcmpFormat.toU8a(), ...receivedMessage.toU8a()];
+    // Send RPC call to inject XCM message
+    // We will set a specific message knowing that it should mint the statemint asset
+    await customWeb3Request(context.web3, "xcm_injectHrmpMessage", [2023, totalMessage]);
+
+    // Create a block in which the XCM will be executed
+    await context.createBlock();
+
+    let status = await context.polkadotApi.query.xcmpQueue.outboundXcmpStatus();
+    expect(status[0].state.isSuspended).to.be.true;
+  });
+
+  it("It should push messages, and should enqueue them in xcmp outbound queue", async function () {
+    // TARGET 100 messages
+    // We want to check there is no visible limit on these
+
+    // Fragments are grouped together and stored in a message
+    // It is this message that we are going to store
+    // The easiest way to test it create a single message every block, with no other messages to append
+
+    // We can create these with sudo
+    // The simplest message should do it
+    const xcmMessage = {
+      V2: [{ ClearOrigin: null as any }],
+    };
+
+    const messageToSend: XcmVersionedXcm = context.polkadotApi.createType(
+      "XcmVersionedXcm",
+      xcmMessage
+    ) as any;
+
+    const destination = {
+      V1: {
+        parents: 1,
+        interior: { X1: { Parachain: 2023 } },
+      },
+    };
+
+    const versionedMult: XcmVersionedMultiLocation = context.polkadotApi.createType(
+      "XcmVersionedMultiLocation",
+      destination
+    ) as any;
+
+    // We also need to trick parachain-system to pretend there exists
+    // an open channel with para id 2023.
+    let paraHrmpMockerTx = mockHrmpChannelExistanceTx(context, 2023, 8, 8192, 4);
+
+    // Test for numMessages
+    let numMessages = 100;
+
+    for (let i = 0; i < numMessages; i++) {
+      await context.createBlock(
```

Comment:
I would check that this `createBlock` here succeeded - either manually via 

```typescript
const { result } = await context.createBlock(...);
expect(result.succeeded).to.be.true;
```

or  via the `expectOk` helper

```typescript
import { expectOk } from "../../util/expect";

await expectOk(context.createBlock(..));
```

This will help us detect the failures in incorrect test setups early.